### ISSUE TITLE
Feature Request: Convenience macro for hour-minute-second quantities

### ISSUE THREAD

**Issue by schtandard** [josephwright/siunitx#684]
A convenience macro for typesetting time quantities consisting of hours minutes and seconds would be nice, analogous to the `\ang` macro. For example `\hms{3;22;15.3}` could output _3 h 22 min 15.3 s_, `\hms{1.5}` _1.5 h_ and `\hms{;15;}` _15 min_.

**Comment by josephwright** [comment]
https://github.com/josephwright/siunitx/issues/70?

**Comment by schtandard** [comment]
Not quite, though they are related. I am asking for a clone of `\ang` for typesetting time. Basically something equivalent to
```latex
\NewDocumentCommand\hms{ O{} m }{%
  \ang[%
    angle-symbol-degree = \hour,
    angle-symbol-minute = \minute,
    angle-symbol-second = \second,
    number-angle-product = \,,
    angle-separator = ~,
    #1,
    ]{#2}%
}
```
though I would expect it to use the value of `quantity-product` instead of a separate `number-angle-product`, at least by default.

This would almost give what #70 is asking for by way of (the analog of) `angle-mode = arc`: With that setting `\hms{1.5}` would give _1 h 30 min_ (well, I am getting a trailing _0 s_, not sure if this is by design). Getting rid of the _min_ with the preceding space could be achieved by tweaking the settings in a somewhat unorthodox way (`angle-symbol-degree = \,\hour, angle-symbol-minute = , number-angle-product = ,`) which is fine, I think. The format _1:30 h_ would not quite be possible (the issue being the trailing _h_).

One could of course go even further and consider an interface for generating arbitrary commands that generate cascading quantities like this, with arbitrary factors between them. This would enable users to create e.g. a command like above but including days as an additional level (`\dhms{2;5;13}` giving _2 d 5 h 13 min_ or _2.217 d_ depending on the settings) or one other non-decimal units like feet and inches or stones and pounds and the like. This would require much more care and work to design an appropriate interface, though, and I don't know if the use-cases justifies this. Also, most of them would probably arise from _not_ using SI units, which is not really what this package is about, I guess.

**Comment by josephwright** [comment]
I think this would be a separate sub-module from angles, or perhaps done the same way as products, etc. (core code the same, different interfaces). I guess the main question for me is the document command: I will look to see what `\hms` is used.

**Comment by josephwright** [comment]
I don't see `\hms` in use in TL, so this looks safe.

**Comment by josephwright** [comment]
OK, I'm looking at this - suggested name is `\duration` or shortened to `\durn`. I am wondering whether to go the same was as I do with 'compound' values (one code path, wrappers for products, ranges, etc.), or whether there is sufficient difference to need two separate but similar  implementations. In particular, whilst for 'compound' values there is the idea they are all 'multi-part', here the relation is more tricky to explain - both have minute/second parts, but the top-level is different. (I could deal with a different conversion factor.)

**Comment by josephwright** [comment]
'Sexagesimal' perhaps?

**Comment by dbitouze** [comment]
> OK, I'm looking at this - suggested name is `\duration` or shortened to `\durn`.

`\duration` is more semantic than `\hms`. IMHO, `\durn` would not be easily understandable.

**Comment by davidcarlisle** [comment]
One problem with `duration` is that an XML schema/ ISO: 8601 `duration` is  optional-sign,year,month,day,hours,minutes,seconds,fractional-seconds so there might be some pressure to extend the value space on the left if it has that name.  

https://www.w3.org/TR/xmlschema-2/#duration

https://en.wikipedia.org/wiki/ISO_8601#Durations

On the other hand `duration is the only one of the types that isn't anchored at a specific actual time.

**Comment by josephwright** [comment]
I'm not sure there's a lot of overlap with XML here :) For a duration of longer than hours, there's really no need to have special formatting as there's very unlikely to be a desire to convert units - this is I imagine something almost exclusively wanted for hours/minutes/seconds.  But I'm not overall keen on `\hms` as a name - it's not really semantic at all.

**Comment by josephwright** [comment]
Better name suggestion: `\timespan`
